service/controller/key/key.go

```diff
@@ -620,6 +620,28 @@ func NodePoolDeploymentName(azureMachinePool *expcapzv1alpha3.AzureMachinePool)
 	return NodePoolVMSSName(azureMachinePool)
 }
 
+func NodePoolEnableSpotInstances(azuremachinePool *expcapzv1alpha3.AzureMachinePool) bool {
+	enabledStr := azuremachinePool.Annotations[AnnotationSpotInstancesEnabled]
+	enabled, err := strconv.ParseBool(enabledStr)
+	if err != nil {
+		// Annotation not found or invalid.
+		return false
+	}
+
+	return enabled
+}
+
+func NodePoolSpotInstancesMaxPrice(azuremachinePool *expcapzv1alpha3.AzureMachinePool) float64 {
+	maxPriceStr := azuremachinePool.Annotations[AnnotationSpotInstancesMaxPrice]
+	maxPrice, err := strconv.ParseFloat(maxPriceStr, 64)
+	if err != nil {
+		// Annotation not found or invalid.
+		return -1
```

Comment:
What happens when Azure is given `maxPrice: -1`?

Reply:
It automatically sets max price to the on demand price.
It's the default. See https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/2019-12-01/virtualmachinescalesets#BillingProfile